luaui/Widgets/gui_pregame_build.lua

```diff
@@ -414,6 +420,12 @@ function widget:DrawWorld()
 		if pos then
 			local bx, by, bz = Spring.Pos2BuildPos(selBuildQueueDefID, pos[1], pos[2], pos[3])
 			local buildFacing = Spring.GetBuildFacing()
+			local offset = 0
+			if (buildFacing % 2 == 1) then
+				offset = (math.abs(UnitDefs[selBuildQueueDefID].zsize - UnitDefs[selBuildQueueDefID].xsize)) * 4
```

Comment:
Longer buildings like the armada air lab is now placed a notable distance from mouse cursor when rotated.